File: Estrutura/Fila.py
class Fila:
    def __init__(self, tamanho_maximo):
        self.tamanho_maximo = tamanho_maximo
        self.fila = []
        self.inicio = 0
        self.fim = -1

    def queue_empty(self):
        return self.fim < self.inicio

    def queue_isFull(self):
        return len(self.fila) == self.tamanho_maximo

    def enqueue(self, valor):
        if not self.queue_isFull():
            self.fila.append(valor)
            self.fim += 1

    def dequeue(self):
        if not self.queue_empty():
            valor = self.fila[self.inicio]
            self.inicio += 1
            return valor

    def print_front(self):
        if not self.queue_empty():
            #return self.fila[self.inicio]
            print(self.fila[self.inicio])

    def print_queue(self):
        if not self.queue_empty():
             for indice in range(self.inicio,self.tamanho_maximo):
                print(self.fila[indice], end=" ")


if __name__ == '__main__':
    fila = Fila(5)
    fila.enqueue(1)
    fila.enqueue(2)
    fila.enqueue(3)
    fila.enqueue(4)
    fila.enqueue(5)
    fila.enqueue(10)

    fila.print_queue();
    print("\n");
    fila.print_front()  # exibe o primeiro elemento da fila (1)

    fila.dequeue()
    fila.print_queue();
    print("\n");
    fila.print_front()  # exibe o primeiro elemento da fila (2)

File: Estrutura/Fila.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 10

int queue[MAX_SIZE];
int inicio = 0;
int fim = -1;
int size = 0;//Tamanho

void enqueue(int value) {
    if (size == MAX_SIZE) {
        printf("Fila Cheia - Queue overflow!\n");
        return;
    }
    //fim = (fim + 1) % MAX_SIZE;
    fim = (fim + 1);

    queue[fim] = value;
    size++;
}

int dequeue() {
    /*if (size == 0) {
        printf("Fila vazia - Queue underflow!\n");
        return -1;
    }*/
    if (inicio > fim) {
        printf("Fila vazia - Queue underflow!\n");
        return -1;
    }
    int value = queue[inicio];
    //inicio = (inicio + 1) % MAX_SIZE;
    inicio = (inicio + 1);

    size--;
    return value;
}

int front()
{
    if(inicio>fim){
        printf("Fila vazia");
        return -1;}
    else
        return queue[inicio];

}

int main() {
    enqueue(1);
    enqueue(2);
    enqueue(3);
    enqueue(4);
    enqueue(5);

    int value = dequeue();
    printf("Valor Retirado: %d\n", value);

    return 0;
}